reclaimer/meta/wrappers/src/byteswapping_ext.c

```diff
@@ -0,0 +1,144 @@
+#include "..\..\..\src\shared.h"
+
+
+static void byteswap_eight_byte_array(
+    char *orig, char *swap, uint64 start, uint64 end, uint64 struct_size,
+    int *offs, int off_count)
+{
+    for (int i = 0; i < off_count; i++) {
+        for (int off = start + offs[i]; off + 8 <= end; off += struct_size) {
+            swap[off]     = orig[off + 7];
+            swap[off + 1] = orig[off + 6];
+            swap[off + 2] = orig[off + 5];
+            swap[off + 3] = orig[off + 4];
+            swap[off + 4] = orig[off + 3];
+            swap[off + 5] = orig[off + 2];
+            swap[off + 6] = orig[off + 1];
+            swap[off + 7] = orig[off];
+        }
+    }
+}
+
+
+static void byteswap_four_byte_array(
+    char *orig, char *swap, uint64 start, uint64 end, uint64 struct_size,
+    int *offs, int off_count)
+{
+    for (int i = 0; i < off_count; i++) {
+        for (int off = start + offs[i]; off + 4 <= end; off += struct_size) {
+            swap[off]     = orig[off + 3];
+            swap[off + 1] = orig[off + 2];
+            swap[off + 2] = orig[off + 1];
+            swap[off + 3] = orig[off];
+        }
+    }
+}
+
+
+static void byteswap_two_byte_array(
+    char *orig, char *swap, uint64 start, uint64 end, uint64 struct_size,
+    int *offs, int off_count)
+{
+    for (int i = 0; i < off_count; i++) {
+        for (int off = start + offs[i]; off + 2 <= end; off += struct_size) {
+            swap[off]     = orig[off + 1];
+            swap[off + 1] = orig[off];
+        }
+    }
+}
+
+
+static PyObject *py_byteswap_struct_array(PyObject *self, PyObject *args) {
+    Py_buffer bufs[2];
+    Py_buffer field_offset_bufs[3];
+    int field_offset_sizes[3] = { 2, 4, 8 };
```

Comment:
This should be a static const outside of the function. That way we're not wasting time allocating stack (or potentially allocating heap) for this array. It also means that a smart compiler can inline it because the values are expected to not change.